Code/Framework/AtomCore/AtomCore/Instance/InstanceId.h

```diff
@@ -52,6 +59,7 @@ namespace AZ
 
             explicit InstanceId(const Uuid& guid);
             explicit InstanceId(const Uuid& guid, uint32_t subId);
+            explicit InstanceId(const Uuid& guid, uint32_t subId, void* data);
```

Comment:
See my other comment
```suggestion
            explicit InstanceId(const Uuid& guid, uint32_t subId, AssetData* assetData);
```